torch/fx/experimental/symbolic_shapes.py

```diff
@@ -2243,16 +2243,19 @@ def _add_guard(self, expr: "sympy.Expr") -> None:
         self.guards.append(guard)
 
     @lru_cache(256)
-    def evaluate_expr(self, expr: "sympy.Expr", hint=None):
+    def evaluate_expr(self, orig_expr: "sympy.Expr", hint=None):
         """
         Given an expression, evaluates it, adding guards if necessary
         """
-        if len(expr.free_symbols) == 0:
-            return expr
-        expr = self.simplify(expr)
+        if len(orig_expr.free_symbols) == 0:
+            log.debug("evaluate_expr %s [trivial]", orig_expr)
+            return orig_expr
+
+        expr = orig_expr
```

Comment:
why? does laundering it through `_maybe_evaluate_static` change it?

Reply:
no, in the future I am going to add a call to simplify here